File: nagasenacs/src/Nagasena/Sax/BinaryDataHandler.cs
using BinaryDataSink = Nagasena.Proc.IO.BinaryDataSink;
using SAXException = Org.System.Xml.Sax.SaxException;

namespace Nagasena.Sax {

  public interface BinaryDataHandler {

    /// <summary>
    /// Mark the start of a binary value.
    /// </summary>
    BinaryDataSink startBinaryData(long totalSize);

    /// <summary>
    /// Writes a binary value where the schema expects a binary value.
    /// </summary>
    void binaryData(byte[] byteArray, int offset, int length, BinaryDataSink binaryDataSink);

    /// <summary>
    /// Mark the end of a binary value.
    /// </summary>
    void endBinaryData(BinaryDataSink binaryDataSink);

  }

}

File: nagasena/src/org/openexi/sax/BinaryDataHandler.java
package org.openexi.sax;

import org.openexi.proc.io.BinaryDataSink;
import org.xml.sax.SAXException;

public interface BinaryDataHandler {

  /**
   * Mark the start of a binary value.
   */
  public BinaryDataSink startBinaryData(long totalSize) throws SAXException;

  /**
   * Writes a binary value where the schema expects a binary value.
   */
  public void binaryData(byte[] byteArray, int offset, int length, BinaryDataSink binaryDataSink) throws SAXException;
  
  /**
   * Mark the end of a binary value.
   */
  public void endBinaryData(BinaryDataSink binaryDataSink) throws SAXException;


}
